lib/host.js

```diff
@@ -63,11 +66,11 @@ Host.prototype.setDown = function() {
     //This host is not down, the pool is being shutdown
     return;
   }
-  this.log('warning', 'Setting host ' + this.address + ' as DOWN');
-  this.unhealthyAt = new Date().getTime();
   this.reconnectionDelay = this.reconnectionSchedule.next().value;
+  this.log('warning', util.format('Host %s considered as DOWN. Reconnection delay %dms.', this.address, this.reconnectionDelay));
+  this.unhealthyAt = new Date().getTime();
```

Comment:
I think using `new Date().getTime()` here and in `canBeConsideredAsUp()` could be a problem since this uses the system clock instead of the realtime clock, so clock skew / time changes could mess things up.  Rarely i've seen that a reconnect is not tried for a host after it fails to connect, i haven't been able to reproduce it viably or figure out why, but I think it may be because of clock skew.   

It is either this or a reconnect is taking less than 1ms, which causes the check in hasTimePassed() to return false since it's a 'greater than' check instead of 'greater than equals'.  I've done some testing locally and i've observed cases where a reconnect is scheduled and then it fails 1001ms after it was submitted (where it's reconnect timer is 1000).

This will probably only ever be a problem in environments where the cassandra node is on the same system as the client.  I think in any case though that is setDown() is called as a result of a failed reconnection we should always schedule another reconnect, unless the host is actually up (reschedule if `unhealthyAt != 0` and failed reconnect maybe?)